examples/fix.kl

```diff
@@ -24,7 +24,7 @@
     length))
 
 
-(define l1 (the (Fix (ListF String)) (in (cons "hello" (in (cons "world" (in (nil))))))))
+(define l1 (the (Fix (ListF (String))) (in (cons "hello" (in (cons "world" (in (nil))))))))
```

Comment:
Types with zero type arguments such as `String` are now spelled `(String)`, just like user-defined datatypes with no type arguments. But only in the kernel; in the prelude #lang, every type is wrapped in a macro which makes the parens optional.

Reply:
Sounds good. Should they be optional or forbidden?

Reply:
`optional-parens` makes them optional rather than mandatory because of data constructors like `nil`, which definitely need parentheses when used as a pattern but not when used as an expression. Currently, types don't suffer from that issue because `type-case` does not yet support recursive patterns. But one day `type-case` will support recursive patterns, and on that day we will be glad that we only made parentheses optional, not forbidden :)